### PR TITLE
N21-1583 copying ctl tools info

### PR DESCRIPTION
**Pull request by mrikallab**
# Short Description

This PR adjust information text in ShareModal, CopyModal and ImportModal, when feature for ctl tools tab is enabled.

## Links to Ticket and related Pull-Requests
https://ticketsystem.dbildungscloud.de/browse/N21-1583

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/copy-result-modal/CopyResultModal.vue [hunk 1]
```diff
@@ -154,6 +163,11 @@ export default {
 				: "";
 			return `${courseFilesText} ${fileErrorText}`.trim();
 		},
+		externalToolsInfoText() {
+			return this.hasFeatureCtlsToolsenabled
```
Comment (IgorCapCoder) [inline src/components/copy-result-modal/CopyResultModal.vue:167]: you already test for the feature flag in the "isShow" property which hangs in an v-if statement in the template. So without the feature flag the whole modal would not be shown. Is there a reason why you would like to do it twice?

File: src/components/share/ImportModal.vue [hunk 2]
```diff
@@ -21,7 +21,17 @@
...
 						<v-icon color="info">{{ mdiInformation }}</v-icon>
 					</div>
-					<div>
+					<div
+						v-if="ctlToolsEnabled"
+						data-testid="import-modal-external-tools-info"
+					>
+						{{
+							$t(
+								`components.molecules.import.courses.options.ctlTools.infoText`
```
Comment (IgorCapCoder) [inline src/components/share/ImportModal.vue:30]: it seems like you could avoid the v-if all together if you would set $(parentType) to "course" and change the language key according to that

File: src/components/copy-result-modal/CopyResultModal.unit.ts [hunk 3]
```diff
@@ -127,6 +135,22 @@ describe("@/components/copy-result-modal/CopyResultModal", () => {
 			);
 		});
 
+		it("should render ctl tools info if root item is a Course and has no failed file ", () => {
+			const copyResultItems = mockResultItems([]);
+			envConfigModule.setEnvs({ FEATURE_CTL_TOOLS_TAB_ENABLED: true } as Envs);
```
Comment (IgorCapCoder) [inline src/components/copy-result-modal/CopyResultModal.unit.ts:140]: what do you think about a test with feature = false

File: src/components/share/ImportModal.unit.ts [hunk 4]
```diff
@@ -122,4 +126,26 @@ describe("@components/share/ImportModal", () => {
 		const emitted = wrapper.emitted("import");
 		expect(emitted).toBeUndefined();
 	});
+
+	describe("when ctl tools are enabled", () => {
```
Comment (IgorCapCoder) [inline src/components/share/ImportModal.unit.ts:130]: with feature = false you would expect $t(`components.molecules.import.${parentType}.options.infoText`). How do you feel about testing that?

File: src/components/share/ShareModal.vue [hunk 5]
```diff
@@ -26,6 +26,16 @@
 							{{ $t(`components.molecules.share.${type}.options.infoText`) }}
 							<br />
 							{{ $t("components.molecules.copyResult.courseFiles.info") }}
+							<div
+								v-if="ctlToolsEnabled"
+								data-testid="share-modal-external-tools-info"
+							>
+								{{
+									$t(
+										`components.molecules.share.courses.options.ctlTools.infotext`
```
Comment (IgorCapCoder) [inline src/components/share/ShareModal.vue:35]: see above: is it possible to use the already provided structure?

**Review by IgorCapCoder — CHANGES_REQUESTED** [review]

File: src/components/copy-result-modal/CopyResultModal.unit.ts (hunk 3 above)
Comment (mrikallab) [reply src/components/copy-result-modal/CopyResultModal.unit.ts:140]: The test of the deactivated feature is above.

File: src/components/share/ImportModal.vue (hunk 2 above)
Comment (mrikallab) [reply src/components/share/ImportModal.vue:30]: I can not avoid the v-if, need to check on the feature flag

**Review by IgorCapCoder — DISMISSED** [review]

**Review by IgorCapCoder — APPROVED** [review]

**Pull request merged**